src/Core/FSpot.FileSystem/GLibDirectory.cs

```diff
@@ -75,6 +75,7 @@ public IEnumerable<SafeUri> Enumerate (SafeUri uri)
 					yield return uri.Append (fileInfo.Name);
 					fileInfo.Dispose ();
 				}
+				fileEnumerator.Close (null);
```

Comment:
Looking at the code this might not be the case. There might be one more place where this is potentially a problem:
https://github.com/mono/f-spot/blob/master/src/Core/FSpot.Core/Photo.cs#L369
```		private bool DirectoryIsEmpty (GLib.File directory)
		{
			uint count = 0;
			GLib.FileEnumerator list = directory.EnumerateChildren ("standard::name",
				GLib.FileQueryInfoFlags.None, null);
			foreach (var item in list) {
				count++;
			}
			return count == 0;
		}
```